sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClient.java

```diff
@@ -0,0 +1,180 @@
+     * @return A {@link PagedIterable} of {@link RoleDefinition role definitions}.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code scope} is null.
+     */
+    public PagedIterable<RoleDefinition> listRoleDefinitions(String scope, String filter, Context context) {
+        return new PagedIterable<>(asyncClient.listRoleDefinitions(scope, filter, context));
+    }
+
+    /**
+     * Get all role definitions that are applicable at scope and above.
+     *
+     * @param scope   The scope of the role definition.
+     * @param context Additional context that is passed through the HTTP pipeline during the service call.
+     * @return A {@link PagedIterable} of {@link RoleDefinition role definitions}.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code scope} is null.
+     */
+    public PagedIterable<RoleAssignment> listRoleAssignments(String scope, Context context) {
+        return listRoleAssignments(scope, null, context);
+    }
+
+    /**
+     * Get all role assignments that are applicable at scope and above.
+     *
+     * @param scope   The scope of the role assignment.
+     * @param filter  The filter to apply on the operation. Use a "atScopeAndBelow" filter to search below the given
+     *                scope as well.
+     * @param context Additional context that is passed through the HTTP pipeline during the service call.
+     * @return A {@link PagedIterable} of {@link RoleAssignment role definitions}.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code scope} is null.
+     */
+    public PagedIterable<RoleAssignment> listRoleAssignments(String scope, String filter, Context context) {
+        return new PagedIterable<>(asyncClient.listRoleAssignments(scope, filter, context));
+    }
+
+    /**
+     * Creates a {@link RoleAssignment}.
+     *
+     * @param scope      The scope of the role assignment to create.
+     * @param name       The name used to create the role assignment.
+     * @param properties Properties for the role assignment.
+     * @return The created {@link RoleAssignment}.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code scope}, {@code name} or {@code properties} are null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public RoleAssignment createRoleAssignment(String scope, String name, RoleAssignmentProperties properties) {
+        return createKeyWithResponse(scope, name, properties, Context.NONE).getValue();
+    }
+
+    /**
+     * Creates a {@link RoleAssignment}.
+     *
+     * @param scope      The scope of the role assignment to create.
+     * @param name       The name used to create the role assignment.
```

Comment:
The service expects a GUID /  UUID for this arg even though it's a string in the swagger. Perhaps we could type constrain it as a hint? We could also consider creating a method that accepts no arg for `name` and generates a new UUID on behalf of the developer.